state/execution.go

```diff
@@ -318,6 +318,32 @@ func getBeginBlockValidatorInfo(block *types.Block, lastValSet *types.ValidatorS
 
 }
 
+func validateValidatorUpdates(abciUpdates []abci.ValidatorUpdate,
+	consensusValidatorParams types.ValidatorParams) error {
+	for _, valUpdate := range abciUpdates {
+		if valUpdate.GetPower() < 0 {
+			return fmt.Errorf("Voting power can't be negative %v", valUpdate)
+		} else if valUpdate.GetPower() == 0 {
+			continue
+		}
+
+		// Check if validator's pubkey matches an ABCI type in the consensus params
+		thisKeyType := valUpdate.PubKey.Type
+		validKeyType := false
+		for i := 0; i < len(consensusValidatorParams.PubKeyTypes); i++ {
+			if consensusValidatorParams.PubKeyTypes[i] == thisKeyType {
```

Comment:
So I think it might be nicer to have this as a method on ValidatorParams, like `ValidatorParams.IsValidPubKeyType`